eng/pipelines/templates/steps/analyze.yml

```diff
@@ -93,7 +93,7 @@ steps:
       TestMarkArgument: ${{ parameters.TestMarkArgument }}
```

Comment:
We also need to be sure to move the artifact publishing to the end of this job and make sure it uses our new upload template so that we can retry this job in the case of failure.